lib/rake-tasks-docker/rake_task.rb

```diff
@@ -41,4 +109,170 @@
+
+    STDOUT.puts "==> docker.env created\n\n"
+  end
+
+  task :copy_dist do |_task, args|
+    Rake::Task['docker-compose.override.yml'].invoke(*args)
+    Rake::Task['docker.env'].invoke(*args)
+  end
+
+  task :setup do |_task, args|
+    Rake::Task['docker:copy_dist'].invoke(*args)
+    Rake::Task['docker:build'].invoke(*args)
+  end
+
+  task :start do |_task, args|
+    Rake::Task['docker:up'].invoke(*args)
+  end
+
+  task :stop do
+    STDOUT.puts '==> Stopping project:'
+    run_process do
+      Process.wait(services_from_args(parse_options).stop)
+      if $?.exitstatus > 0
+        STDERR.puts "==> The project failed to stop\n\n"
+        exit(1)
+      end
+    end
+    STDOUT.puts "==> Project stopped\n\n"
+  end
+
+  task :restart do |_task, args|
+    Rake::Task['docker:stop'].invoke(*args)
+    Rake::Task['docker:start'].invoke(*args)
+  end
+
+  task :destroy do |_task, args|
+    Rake::Task['docker:stop'].invoke(*args)
+    STDOUT.puts '==> Removing containers and volumes for project:'
+    run_process do
+      Process.wait(services_from_args(parse_options).down)
+      if $?.exitstatus > 0
+        STDERR.puts "==> The project failed to be destroyed\n\n"
+        exit(1)
+      end
+    end
+    STDOUT.puts "==> Project containers and volumes removed\n\n"
+  end
+
+  task :reset do |_task, args|
+    Rake::Task['docker:destroy'].invoke(*args)
+    Rake::Task['docker:build'].invoke(*args)
+    Rake::Task['docker:start'].invoke(*args)
+  end
+
+  task :ip do
+    options = parse_options
+    services = options[:services]
+    if services.size > 1
+      STDOUT.puts services_from_args(options).ip.to_json
+    elsif services.size == 1
+      STDOUT.puts services_from_args(options).ip[services.first]
+    else
+      STDERR.puts "==> Please provide a service to look up the IP for\n\n"
+      exit(1)
+    end
+  end
+
+  task :logs do
+    options = parse_options
+    services_from_args(options).logs(false)
+  end
+
+  task :command do
+    command_args = parse_command
+    services_from_args(command_args).exec(command_args[:user], command_args[:command])
+  end
+
+  task :bash do
+    args = parse_argv
+    args << "--command=bash"
+    command_args = parse_command(args, 'web')
+    services_from_args(command_args).exec(command_args[:user], command_args[:command])
+  end
+
+  task :hostsfile do
+    options = parse_hostname
+    services = options[:services]
+    hostname = options[:hostname]
+    if services.length != 1
+      STDERR.puts "==> Please specify only one service for docker:hostsfile\n\n"
+      exit(1)
+    end
+    service = services.first
+    STDOUT.puts '==> Adding hostname to hosts'
+    ip = services_from_args(options).ip[service]
+    unless ip
+      STDERR.puts "==> Failed to find the IP address of the #{service} container\n\n"
+      exit(1)
+    end
+    hosts_entry = "#{ip} #{hostname}"
+    current_hosts = File.read('/etc/hosts')
+    if current_hosts =~ /^#{Regexp.escape(hosts_entry)}$/
```

Comment:
This doens't work if the hostsfile has entries for previous boot-ups on linux where the container IP can change - e.g. `172.18.0.2 example.test` existing but `172.18.0.4 example.test` is being checked for.